core/common/src/EntityProps.ts

```diff
@@ -83,6 +83,7 @@ export enum PrimitiveTypeCode {
   Point2d = 0x701, // eslint-disable-line @typescript-eslint/no-shadow
   Point3d = 0x801, // eslint-disable-line @typescript-eslint/no-shadow
   String = 0x901,
+  IGeometry = 0xa01,
```

Comment:
Can you document this? The other ones are obvious, but people (including me) don't really know what "IGeometry" is.

Reply:
@pmconne - Done.

Reply:
Thanks - but does anyone (especially outside of Bentley) know what "Bentley.Geometry.Common.IGeometry" means? That's a C# namespace, isn't it?

Reply:
Seems we have a broader documentation problem with this type.
- ["a common geometry value of any type"](https://www.itwinjs.org/presentation/advanced/ecexpressions/#other-primitive-types)
- ["2561"](https://www.itwinjs.org/bis/domains/ecdbmeta.ecschema/#primitivetype)
- ["See Common Geometry documentation for more information on individual types"](https://www.itwinjs.org/bis/ec/primitive-types/) - no indication of where to find such documentation.
- [A "full list" link that produces a 404](https://www.itwinjs.org/bis/guide/references/class-diagram-conventions/#content)

@ColinKerr what can consumers of iTwin.js do with properties of IGeometry properties? There are a whole bunch in our domain schemas.

Reply:
We also have a `PrimitiveType` enum in ecschema-metadata, which does include `IGeometry` (undocumented). This `PrimitiveTypeCode` enum may have originated from the one in ECValue.cpp, which also omits `IGeometry`.

Reply:
> Thanks - but does anyone (especially outside of Bentley) know what "Bentley.Geometry.Common.IGeometry" means? That's a C# namespace, isn't it?

Bentley.Geometry.Common.IGeometry is the C# namespace, but IGeometry types are also used in native code. That is the base type for all native geometry types: ICurvePrimitive, CurveVector, ISolidPrimitive, MSBsplineSurface, PolyfaceHeader.  I could change the documentation to that instead.  I was going with the XML documentation which uses 'Bentley.Geometry.Common.IGeometry' to refer to both languages.

Reply:
Thanks @c-macdonald. This is a bigger question than your simple PR, that we need someone like @ColinKerr to answer: what if anything can users of iTwin.js expect to do with IGeometry types, and what documentation can we provide to enable them to do those things?

Question directly relevant to your PR, also probably for @ColinKerr, is why do we have `PrimitiveType` and `PrimitiveTypeCode`, the latter of which omits `IGeometry`, and do we really need both?

Reply:
> We also have a `PrimitiveType` enum in ecschema-metadata, which does include `IGeometry` (undocumented). This `PrimitiveTypeCode` enum may have originated from the one in ECValue.cpp, which also omits `IGeometry`.

Yes, there are inconsistencies in where IGeometry is implemented. My code was only looking at PrimitiveTypeCode, so that's where I knew it needed to get fixed.  The native C++ code uses: json[json_primitiveType()] = prop.GetType(); which _does_ implement IGeometry. So the code will set a value that typescript then can't read.

Reply:
It is very unfortunate that the old C# namespace is the type name in BIS schemas, but I think that is find so long as we fix up the documentation.

Folks can work with IGeometry values using the iTwin.js geometry libraries.  I cannot find good docs for this but there are tests here: https://github.com/iTwin/itwinjs-core/blob/master/core/backend/src/test/element/ElementRoundTrip.test.ts

What link is 404, this link works for me, it sends me here https://www.itwinjs.org/bis/ec/primitive-types/
![image](https://user-images.githubusercontent.com/33028649/199007857-2fa8ee82-52c9-466e-8e4b-f63a35b459c3.png)

We need to create a new documentation issue to go through and clean this all up.

Reply:
> Question directly relevant to your PR, also probably for @ColinKerr, is why do we have PrimitiveType and PrimitiveTypeCode, the latter of which omits IGeometry, and do we really need both?

Reply:
We have two parallel metadata implementations, one in core-backend and one in ecschema-metadata.  The one in ecschema-metadata is more 'complete'  and designed to run in the browser and in node.  The original goal was to merge the two but that didn't make it into 1.0 and it has not been high enough priority to get scheduled for 2.0 or 3.0.  It would be a breaking change and while it would make for cleaner API it would not provide any other user value.

So for at least the 3.x generation we will need both.